staging/src/k8s.io/client-go/openapi/cached/groupversion.go

```diff
@@ -17,17 +17,33 @@ limitations under the License.
 package cached
 
 import (
+	"encoding/json"
 	"sync"
 
 	openapi_v3 "github.com/google/gnostic/openapiv3"
+	"google.golang.org/protobuf/proto"
 	"k8s.io/client-go/openapi"
+	"k8s.io/kube-openapi/pkg/spec3"
 )
 
 type groupversion struct {
 	delegate openapi.GroupVersion
-	once     sync.Once
-	doc      *openapi_v3.Document
-	err      error
+
+	docOnce sync.Once
+	doc     *openapi_v3.Document
+	docErr  error
+
+	kubeOnce sync.Once
+	kubeDoc  *spec3.OpenAPI
+	kubeErr  error
+
+	jsonOnce  sync.Once
+	jsonBytes []byte
+	jsonErr   error
+
+	pbOnce  sync.Once
+	pbBytes []byte
+	pbErr   error
```

Comment:
That's a little yucky but I'm not exactly sure what to propose here ...

Reply:
thought about doing a generic `Lazy[T]` for this purpose. not sure how generics policy affects this type, though